sky/global_user_state.py

```diff
@@ -104,19 +116,47 @@ class StorageStatus(enum.Enum):
     READY = 'READY'
 
 
-def add_or_update_cluster(cluster_name: str,
-                          cluster_handle: 'backends.Backend.ResourceHandle',
-                          ready: bool,
-                          is_launch: bool = True):
+def add_or_update_cluster(
+    cluster_name: str,
+    cluster_handle: 'backends.Backend.ResourceHandle',
+    requested_resources: Optional[Set[Any]],
+    ready: bool,
+    is_launch: bool = True,
+):
     """Adds or updates cluster_name -> cluster_handle mapping."""
     # FIXME: launched_at will be changed when `sky launch -c` is called.
     handle = pickle.dumps(cluster_handle)
     cluster_launched_at = int(time.time()) if is_launch else None
     last_use = common_utils.get_pretty_entry_point() if is_launch else None
     status = ClusterStatus.UP if ready else ClusterStatus.INIT
+
+    cluster_hash = _get_cluster_hash(cluster_name) or str(uuid.uuid4())
+
+    def get_launched_resources_from_handle(
+            handle: Optional['backends.Backend.ResourceHandle']):
+        launched_resources = handle.launched_resources
+        launched_nodes = handle.launched_nodes
+        return (launched_nodes, launched_resources)
+
+    launched_resources = get_launched_resources_from_handle(cluster_handle)
+
+    usage_intervals = _get_cluster_usage_intervals(cluster_hash)
+
+    if requested_resources and not usage_intervals:
```

Comment:
Add a comment for this condition? It is a bit unclear in what case will the condition holds (e.g. the cluster is firstly launched) at the first glance.